### PR TITLE
Flatten Binary Tree to Linked List

### PR DESCRIPTION
**Pull request by shivamtrip-st**
## Related Issue
- Flattening of binary tree to linked list    ( #536)

## Proposed Changes

-  uploaded code file of flattening  of binary tree to linked list.

- our code gives inorder traversal of binary tree to show the result.

## Additional Info
-  convert binary tree such that , where the right child pointer points to the next node in the list and the left child pointer is always null .gives us required result.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [ ] 📝 My change requires a change to the README.
- [ ] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [ ] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

### SOME DETAILS ABOUT OUR PROGRAM.

- solution is basically doing is putting the right subtree next to the rightmost node on the left subtree and then making the left subtree the right subtree and then making the left one null.
- 
image

- here we have used morris algorithm to flatten the binary tree.

as it is effiecient ,the time and space complexity of our code is  :

- Time complexity :O(n)

- Space complexity : O(1)

## Output Screenshots
| Screenshot  1      | Screenshot 2  |
| ----------- | ----------- |
| Given Input| Required Output     |
| 
Screenshot (180)
  |  
Screenshot (181)
    |

### CODE REVIEW

**Comment by shivamtrip-st** [comment]
please review @Lakhankumawat @siddhi-244

File: T-Tree/T-TreeTraversals/flattening a binary tree.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  
+  //creating new tree node
+TreeNode* newTreeNode(int value)
+{TreeNode* root=new TreeNode;
+root->val=value;
+root->right=root->left=nullptr;
+	return root;
+}
+
+//morris algorithm
```
Comment (Lakhankumawat) [inline T-Tree/T-TreeTraversals/flattening a binary tree.cpp:22]: https://github.com/Lakhankumawat/LearnCPP/blob/main/T-Tree/T-TreeTraversals/Morris_Traversal.cpp

File: T-Tree/T-TreeTraversals/flattening a binary tree.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                //Use current node's left subtree to replace its right subtree(original right 
+                //subtree is already linked by current node's prenode
+				now->right = now->left;
+				now->left = nullptr;
+			}
+			now = now->right;
+		}
+    }
+    //inorder traversal
+void inorder(TreeNode *root)
```
Comment (Lakhankumawat) [inline T-Tree/T-TreeTraversals/flattening a binary tree.cpp:45]: https://github.com/Lakhankumawat/LearnCPP/blob/main/T-Tree/T-TreeTraversals/inorderTraversal.cpp

**Review by Lakhankumawat — COMMENTED** [review]
I don't see anything new in your code , and it is not a valid contribution according to me .

**Comment by Lakhankumawat** [comment]
1. Not Starred The Repo but checked in Checklist
2. Description is good
3. NO README

**Comment by siddhi-244** [comment]
Closing this pr

**Pull request closed without merging**